### PR TITLE
Apply more restrictive SecurityContext to components

### PR DESCRIPTION
**Pull request by hjiawei**
## Description

This changeset add more restrictive SecurityContext to Calico opensource
and enterprise components. Least privilege principle is applied to this
change.

Data plane components are mostly running as root user and application
layer compoennts are running as non-root users. `Capabilities` and
`SeccompProfile` are also filled with sensible defaults in
`SecurityContext`s.

## For PR author

- [x] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/egressgateway/egressgateway.go [hunk 1]
```diff
@@ -169,17 +172,23 @@ func (c *component) egwBuildAnnotations() map[string]string {
 }
 
 func (c *component) egwInitContainer() *corev1.Container {
+	sc := securitycontext.NewRootContext(true)
```
Comment (hjiawei) [inline pkg/render/egressgateway/egressgateway.go:175]: Pending validation from @sridhartigera.

**Review by rene-dekker — APPROVED** [review]
lgtm

File: pkg/render/logstorage.go [hunk 2]
```diff
@@ -584,6 +577,16 @@ func (es elasticsearchComponent) podTemplate() corev1.PodTemplateSpec {
...
 
+	sc := securitycontext.NewRootContext(false)
+	// These capabilities are required for docker-entrypoint.sh.
+	// See: https://github.com/elastic/elasticsearch/blob/7.17/distribution/docker/src/docker/bin/docker-entrypoint.sh.
+	// TODO Consider removing for Elasticsearch v8+.
+	sc.Capabilities.Add = []corev1.Capability{
+		"SETGID",
+		"SETUID",
+		"SYS_CHROOT",
+	}
```
Comment (anthonytwh) [inline pkg/render/logstorage.go:588]: Does the ES entrypoint script work without `SETGID` and `SETUID`?

Comment (hjiawei) [reply pkg/render/logstorage.go:588]: Not really. I believe `chroot` needs them.

**Comment by hjiawei** [comment]
I have tested a standard Calico Enterprise install on the following platforms:

- gcp-kubeadmin
- gcp-mke
- gcp-rancher
- gcp-rke2 (cis not enabled)
- aws-eks
- aws-openshift
- azr-aks

**Pull request merged**